### ISSUE TITLE
Review the .travis.yaml file from `cardano-wallet` (+ connect travis to new repository)

### ISSUE THREAD

**Issue by piotr-iohk** [cardano-foundation/cardano-wallet#1]
# Context
We want to have CI set up for the new `cardano-wallet` in order to have a means to automatically build the code and run all our tests against the build. At this moment we decided to go with Travis, so we want to get the `.travis.yaml` from the previous `cardano-wallet` review it and remove all unnecessary dependencies to `cardano-sl` from it. This task is also to make sure that the new repo is connected to Travis CI.

# Decision

Connect new `cardano-wallet` repo with Travis CI. Get the `.travis.yaml` from the previous `cardano-wallet` review it and remove all unnecessary dependencies to `cardano-sl` from it. This task is also to make sure that the new repo is connected to Travis CI.

# AC

- [ ] Travis is running builds on PRs.
- [ ] Travis is running builds on the master branch.
- [ ] Tests are being executed by Travis.

---

# Development Plan

- [x] Initial Travis build which runs `stack test`.
- [x] Connect repo to travis.
- [x] Also run Coveralls and stylish-haskell.

# PR

| Number   | Base      |
| ---      | ---       |
| #15       | `master`  |
| #19  | `master` | 

# QA 

- Check that the list of AC is complete and done.

**Comment by rvl** [comment]
We also have https://github.com/jonathanknowles/haskell-travis-example

**Comment by KtorZ** [comment]
I'll also add support for stylish-haskell & coveralls :+1:

**Comment by piotr-iohk** [comment]
lgtm :+1: 

I think it would be nice to have stylish-haskell, weeder, hlint, tests as separate checks in Travis for better readibility (as it was on previous repo). I think also the checks are then capable of runnig in parallel. 
Perhaps it is not needed now but I will add that as a thought to the wiki for discussion.

**Comment by KtorZ** [comment]
Agreed @piotr-iohk , and ideally, we would even push it further and have the status reported inside github directly as separate CI checks. @Anviking had some ideas on the matter :)
